File: Data Structures & Algorithms/subsets-ii/submission-5.cpp
class Solution {
public:
    vector<vector<int>> res;
    vector<vector<int>> subsetsWithDup(vector<int>& nums) {
        vector<int> curr;
        sort(nums.begin(), nums.end());
        dfs(nums, curr, 0);
        return res;
    }
    void dfs(vector<int>&nums, vector<int> & curr, int i){
        if(find(res.begin(), res.end(), curr) == res.end()) 
        res.push_back(curr);
        if(i == nums.size()) return;
        for(int j = i; j < nums.size(); j++){
            curr.push_back(nums[j]);
            dfs(nums, curr, j+1);
            curr.pop_back();
        }
    }
};

File: Data Structures & Algorithms/max-consecutive-ones/submission-5.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:

        prev = 2
        res, count = 0, 0
        for i in nums:
            if i == prev and i == 1:
                count +=1
            elif i == 1:
                res = max(count, res)
                count = 1
            prev = i
        res = max(count, res)
        return res